### PR TITLE
Sauce Connect and other config setting clarification

### PR DESCRIPTION
**Pull request by sweeneyskirt-sl**
### Description
- Edited Configuration Getting Started page and other pages affected by the configuration edits.

### Motivation and Context
Testers requested clarification on how Sauce Connect works with saucectl, so I added a section to the Configuration page to describe the behavior and associated settings. In so doing, I also added some information about setting up a proxy to get it out of the FAQ, which I would ultimately like to retire because it has a lot of outdated info in there and what is there is already represented on other pages, with the exception of the links to the docker images. Perhaps we should add these to the Docker supported frameworks table?

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [ ] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/master/CONTRIBUTING.MD) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/testrunner-toolkit/configuration.md [hunk 1]
```diff
@@ -1,177 +1,221 @@
...
+```jsx title= "config.yml npm example"
+npm:
+  registry: https://registry.npmjs.org
+  packages:
+    lodash: "4.17.20"
+    "@babel/preset-typescript": "7.12"
+    "@cypress/react": "^5.0.1"
+```
+:::note
+The `registry` parameter is only supported in Cypress running in Docker.
```
Comment (tianfeng92) [inline docs/testrunner-toolkit/configuration.md:220]: The `registry` parameter is supported in `cypress`, `playwright` and `testcafe` runner. And it works for sauce cloud as well for these 3 runners.

File: docs/testrunner-toolkit/faqs.md [hunk 2]
```diff
@@ -36,55 +29,15 @@ To learn more about:
 All images are hosted on Docker Hub.
 
 * [Base image](https://hub.docker.com/r/saucelabs/testrunner-image/tags)
-is called `testrunner`. It contains the tooling necessary to record videos, VNC etc. Plus Chrome, and a Firefox version.
-
+is called `testrunner`. It contains the tooling necessary to record videos, VNC etc., plus Chrome, and a Firefox version.
 * [Base image + Playwright](https://hub.docker.com/r/saucelabs/stt-playwright-jest-node/tags)
-contains saucectl with different versions of Playwright.
-
+contains `saucectl` with different versions of Playwright.
```
Comment (tianfeng92) [inline docs/testrunner-toolkit/faqs.md:34]: cypress/playwright/testcafe/puppeteer runners no longer contain `saucectl`. suggest to remove saucectl here.

File: docs/testrunner-toolkit/faqs.md [hunk 3]
```diff
@@ -36,55 +29,15 @@ To learn more about:
...
 * [Base image + Puppeteer](https://hub.docker.com/r/saucelabs/stt-puppeteer-jest-node/tags)
-contains saucectl with different versions of Puppeteer.
-
+contains `saucectl` with different versions of Puppeteer.
```
Comment (tianfeng92) [inline docs/testrunner-toolkit/faqs.md:36]: same above. should remove `saucectl`

File: docs/testrunner-toolkit/faqs.md (hunk 2 above)
Comment (tianfeng92) [reply docs/testrunner-toolkit/faqs.md:34]: And playwright runner docker page should be this 
```
https://hub.docker.com/r/saucelabs/stt-playwright-node/tags
```

File: docs/testrunner-toolkit/faqs.md [hunk 4]
```diff
@@ -36,55 +29,15 @@ To learn more about:
...
 * [Base image + TestCafe](https://hub.docker.com/r/saucelabs/stt-testcafe-node/tags)
-contains saucectl with different versions of TestCafe.
-
-* [Base image + Cypress](https://hub.docker.com/r/saucelabs/stt-cypress-mocha-node/tags) contains saucectl with different versions of Cypress.
+contains `saucectl` with different versions of TestCafe.
```
Comment (tianfeng92) [inline docs/testrunner-toolkit/faqs.md:38]: same above. should remove `saucectl`

File: docs/testrunner-toolkit/faqs.md [hunk 5]
```diff
@@ -36,55 +29,15 @@ To learn more about:
...
+* [Base image + Cypress](https://hub.docker.com/r/saucelabs/stt-cypress-mocha-node/tags) contains `saucectl` with different versions of Cypress.
```
Comment (tianfeng92) [inline docs/testrunner-toolkit/faqs.md:39]: same above. should remove `saucectl`.

File: docs/testrunner-toolkit/configuration.md (hunk 1 above)
Comment (sweeneyskirt-sl) [reply docs/testrunner-toolkit/configuration.md:220]: Thanks!

File: docs/testrunner-toolkit/faqs.md (hunk 2 above)
Comment (sweeneyskirt-sl) [reply docs/testrunner-toolkit/faqs.md:34]: Thanks @tianfeng92. I am considering moving these docker hub image links to the docker supported frameworks table on the SC home page and deleting the FAQ page altogether. What do you think?

Comment (tianfeng92) [reply docs/testrunner-toolkit/faqs.md:34]: @sweeneyskirt-sl agree. FAQ seems to be redundant after most of documentations added

File: docs/testrunner-toolkit/installation.md [hunk 6]
```diff
@@ -119,7 +119,7 @@ SAUCE_ACCESS_KEY='valid.key'
 ```
 
 :::warning Protect your Credentials
-Whether you are using environment variables or a credentials file, make sure your authentication data is protected. Use secrets or context variables to mask your environment variables, or add `credentials.yml` to your `gitignore` file to ensure your credentials are not exposed in your commits.
+Whether you are using environment variables or a credentials file, make sure your authentication data is protected. Use secrets or context variables to mask your environment variables, or add `credentials.yml` to your `sauceignore` file to ensure your credentials are not exposed in your commits.
```
Comment (alexplischke) [inline docs/testrunner-toolkit/installation.md:122]: This should indeed still be the `.gitignore` file! The `.gitignore` file is git's way of exluding files from being committed to the repo ⚠️

File: docs/testrunner-toolkit/configuration.md [hunk 7]
```diff
@@ -1,177 +1,221 @@
...
+```
+
+This is the identifer `saucectl` expects as the `id` value in the config file, even though the Sauce Labs UI refers to this values as the `Tunnel Name`.
+:::
+
+## Setting up a Proxy
+
+If you need to go through a proxy server, you can set it through the following variables:
+
+* `HTTP_PROXY`: Proxy to use to access HTTP websites
```
[the remark this answers is not in the document]
Comment (alexplischke) [reply docs/testrunner-toolkit/configuration.md:74]: Might be worth adding a note/warning/disclaimer/hint (whatever that info box is called :D) to point the known issue with playwright out. Until we sort it out with the playwright folks :)

File: docs/testrunner-toolkit/faqs.md [hunk 8]
```diff
@@ -1,25 +1,18 @@
...
-## How Does Testrunner Toolkit Work?
-Native JavaScript testing is achieved through the combination of Sauce Labs, Jest, and the
-JavaScript framework of your choice. In the current beta, the toolkit supports:
+## How Does Sauce Control Work?
 
-* [Puppeteer](https://github.com/puppeteer/puppeteer)
-* [Playwright](https://github.com/microsoft/playwright)
-* [TestCafe](https://github.com/DevExpress/testcafe)
-* [Cypress](https://github.com/cypress-io/cypress)
+Native testing is achieved through the combination of Sauce Labs, Jest, and the framework of your choice, giving you the power and expressiveness of different test frameworks with the dashboards, infrastructure, and analytics of [Sauce Labs](https://saucelabs.com/).
```
Comment (alexplischke) [inline docs/testrunner-toolkit/faqs.md:11]: `Jest` is only really used as part of puppeteer now.
Cypress uses `mocha`, playwright uses its own flavor `playwright-test`, testcafe also has its own.
So I'd suggest simply to skip mentioning Jest in this context :D

File: docs/testrunner-toolkit/installation.md [hunk 9]
```diff
@@ -131,7 +131,7 @@ This process generates a set of dependencies that allow you to quickly set up a
 * a framework directory (e.g., `cypress`) and other relevant files (e.g., `cypress.json`)
 * an example test for the chosen framework (e.g., `cypress/integration/example.test.js`)
 
-If you already have working tests in your framework, you can skip this section and [run your tests](testrunner-toolkit/running-tests) in order to run your existing tests.
+If you already have working tests in your framework, you can skip this section and [run your tests](/running-tests) in order to run your existing tests.
```
Comment (sweeneyskirt-sl) [inline docs/testrunner-toolkit/installation.md:134]:
```suggestion
If you already have working tests in your framework, you can skip this section and [run your tests](/testrunner-toolkit/running-tests) in order to run your existing tests.
```

**Pull request merged**